test/test_tags.rb

```diff
@@ -13,7 +13,7 @@ def create_post(content, override = {}, converter_class = Jekyll::Converters::Ma
     site = Site.new(Jekyll.configuration)
 
     if override['read_posts']
-      site.read_posts('')
+      site.posts += PostReader.new(site).read('')
```

Comment:
In the context of optimization (as you implied) that matters very little to Jekyll (as in the type of app we are not the type of programmers we are but, really, it should be applied to both.)  We are a short lived process and our GC  time matters little  we probably don't even hit the exponential scale required of caring about the GC time and whether or not we are creating new array's and marking old objects.